changes_revision.go

```diff
@@ -445,8 +475,17 @@ func (s *ChangesService) GetPatch(ctx context.Context, changeID, revisionID stri
 		return nil, nil, err
 	}
 
-	v := new(string)
-	resp, err := s.client.Do(req, v)
+	strVal := ""
+	var v *string = &strVal
+
+	// Create an instance of our writer struct since the /patch endpoint
+	// returns a base64 encoded string which cannot be marshalled as JSON.
+	stringWriter, err := NewStringPointerWriter(v)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	resp, err := s.client.Do(req, stringWriter)
```

Comment:
I suspect it's better to go with another way to achieve the outcome you want.

Consider the similar GetCommitContent method which uses `getStringResponseWithoutOptions`:

https://github.com/andygrunwald/go-gerrit/blob/31687dc5673cabc520980d8b1de2fe4d8f7cf666/projects_commit.go#L49-L56

As another example, see the approach used by another Gerrit client library here:

https://cs.opensource.google/go/x/build/+/1810ea16:gerrit/gerrit.go;l=904-912;drc=2a0314b810a580bd009db088d0d17ba343b2e5ae